### PR TITLE
Deprecate auto_mode

### PR DESCRIPTION
**Pull request by dwalton76**

### CODE REVIEW

**Comment by dwalton76** [comment]
For issue #378

File: ev3dev/core.py [hunk 1]
```diff
@@ -2033,12 +2035,26 @@ def mode(self):
...
 
     @mode.setter
     def mode(self, value):
+        """
+        This will be called often so we cache the current mode in
+        self._mode_value to avoid some file IO.
+        """
+
+        # If 'mode' is already set to 'value' there is nothing to do.
+        if value == self._mode_value:
```
Comment (WasabiFan) [inline ev3dev/core.py]: Do we want a way to bypass this cache logic? Previously, the sysfs accessors were always guaranteed to perform a write operation, whereas now there's no way to handle the case where you intentionally want multiple processes interfacing with the same device. This is a result of the fact that caching was moved down a level from the friendly functions to the getters/setters.

Comment (dwalton76) [reply ev3dev/core.py]: Part of me is wondering if we need it at all.  We didn't have it before and I've never noticed any perf issues with setting the modes.  We aren't setting the mode any more often with the deprecation of auto_mode (assuming nobody ever set this to false).  Maybe we just chop the `mode_value`/caching part of this patch?

Comment (dwalton76) [reply ev3dev/core.py]: @WasabiFan what say you on the `mode_value`...keep or chop?  I'm leaning towards chop and address it as part of #365

Comment (WasabiFan) [reply ev3dev/core.py]: Agreed. I don't like the inconsistency this introduces, especially in light of cases such as the ultrasonic sensor where the only way to send a ping is to set the mode again. We should cut it from here and address it elsewhere.

**Comment by dwalton76** [comment]
Removed `_mode_value`, merging

**Pull request merged**